packages/common/auth/src/CrossmintAuth.ts

```diff
@@ -0,0 +1,49 @@
+import { type Crossmint, CrossmintApiClient } from "@crossmint/common-sdk-base";
+import { CROSSMINT_API_VERSION, SDK_NAME, SDK_VERSION } from "./utils/constants";
+import { type AuthMaterialWithUser, CrossmintAuthenticationError, type AuthMaterialResponse } from "./types";
+
+export class CrossmintAuth {
+    protected crossmint: Crossmint;
+    protected apiClient: CrossmintApiClient;
+
+    constructor(crossmint: Crossmint) {
```

Comment:
if this is meant to be private, I'd also add `CrossmintApiClient` as a dependency and move creation of that object to the `from` method.

Reply:
Maybe this was designed to be extended?
If so, I'd suggest redesigning to favor composition over inheritance. Let me know if I can help with that!

Reply:
Open to suggestions. Here I implemented as per the approved PRD and we discussed multiple API surfaces and ended up going with this one. So following this `from` approach is a must. But if there's anything we can do to improve the implementation, let me know!

Reply:
I'm not against the static `from` method. I'm just not sure why we'd have inheritance as the extension mechanism.